cmd/kube-controller-manager/app/core.go

```diff
@@ -294,16 +293,18 @@ func startResourceQuotaController(ctx ControllerContext) (bool, error) {
 }
 
 func startNamespaceController(ctx ControllerContext) (bool, error) {
-	// TODO: should use a dynamic RESTMapper built from the discovery results.
-	restMapper := legacyscheme.Registry.RESTMapper()
-
 	// the namespace cleanup controller is very chatty.  It makes lots of discovery calls and then it makes lots of delete calls
 	// the ratelimiter negatively affects its speed.  Deleting 100 total items in a namespace (that's only a few of each resource
 	// including events), takes ~10 seconds by default.
 	nsKubeconfig := ctx.ClientBuilder.ConfigOrDie("namespace-controller")
 	nsKubeconfig.QPS *= 10
 	nsKubeconfig.Burst *= 10
 	namespaceKubeClient := clientset.NewForConfigOrDie(nsKubeconfig)
+
+	// Use a dynamic restMapper.
+	discoveryClient := cacheddiscovery.NewMemCacheClient(namespaceKubeClient.Discovery())
```

Comment:
Please add this to the controller context and plumb it to all locations.

Reply:
make sense